bolt-modules/boltlib/lib/puppet/functions/run_plan.rb

```diff
@@ -117,7 +117,7 @@ def run_inner_plan(scope, plan_name, targets, args = {})
         # undef/nil
         result = catch(:return) do
           scope.with_global_scope do |global_scope|
-            closure.call_by_name_with_scope(global_scope, params, true)
+            executor.run_plan(global_scope, closure, params)
```

Comment:
@McdonaldSeanp I think this will help you longterm too but heads up.

Reply:
I finally got around to taking a look. Thanks for the heads up, we'll make sure `run_plan` is available in PE's executor before we pull in the bolt update with this.

Long term I think you're right that this should be helpful, because the PE executor can intercept `run_plan` calls now, which allows us to make it more stable from the PE side.